VirsTimer.Server/virstimer/src/main/kotlin/pl/virstimer/service/ScrambleService.kt

```diff
@@ -12,20 +12,143 @@ class ScrambleService {
+    }
+
     private fun scrambleThreeByThree(): DomainScramble {
         val puzzle = ThreeByThreeCubePuzzle()
         val wcaScramble = puzzle.generateWcaScramble(random)
 
         return DomainScramble(
-                wcaScramble,
-                puzzle.drawScramble(wcaScramble, puzzle.defaultColorScheme),
-                PuzzleType.THREE_BY_THREE
+            wcaScramble,
+            puzzle.drawScramble(wcaScramble, puzzle.defaultColorScheme),
+            PuzzleType.THREE_BY_THREE
+        )
+    }
+
+    private fun scrambleFourByFour(): DomainScramble {
+        val puzzle = FourByFourCubePuzzle()
+        val wcaScramble = puzzle.generateWcaScramble(random)
+
+        return DomainScramble(
+            wcaScramble,
+            puzzle.drawScramble(wcaScramble, puzzle.defaultColorScheme),
+            PuzzleType.FOUR_BY_FOUR
+        )
+    }
+
+    private fun scrambleFiveByFive(): DomainScramble {
+        val puzzle = NoInspectionFiveByFiveCubePuzzle()
+        val wcaScramble = puzzle.generateWcaScramble(random)
+
+        return DomainScramble(
+            wcaScramble,
+            puzzle.drawScramble(wcaScramble, puzzle.defaultColorScheme),
+            PuzzleType.FIVE_BY_FIVE
+        )
+    }
+
+    private fun scrambleSixBySix(): DomainScramble {
+        val puzzle = CubePuzzle(6)
+        val wcaScramble = puzzle.generateWcaScramble(random)
+
+        return DomainScramble(
+            wcaScramble,
+            puzzle.drawScramble(wcaScramble, puzzle.defaultColorScheme),
+            PuzzleType.SIX_BY_SIX
+        )
+    }
+
+    private fun scrambleSevenBySeven(): DomainScramble {
+        val puzzle = CubePuzzle(7)
+        val wcaScramble = puzzle.generateWcaScramble(random)
+
+        return DomainScramble(
+            wcaScramble,
+            puzzle.drawScramble(wcaScramble, puzzle.defaultColorScheme),
+            PuzzleType.SEVEN_BY_SEVEN
+        )
+    }
+
+    private fun scrambleClock(): DomainScramble {
+        val puzzle = ClockPuzzle()
+        val wcaScramble = puzzle.generateWcaScramble(random)
+
+        return DomainScramble(
+            wcaScramble,
+            puzzle.drawScramble(wcaScramble, puzzle.defaultColorScheme),
+            PuzzleType.CLOCK
+        )
+    }
+
+    private fun scrambleMegaminx(): DomainScramble {
+        val puzzle = MegaminxPuzzle()
+        val wcaScramble = puzzle.generateWcaScramble(random)
+
+        return DomainScramble(
+            wcaScramble,
+            puzzle.drawScramble(wcaScramble, puzzle.defaultColorScheme),
+            PuzzleType.MEGAMINX
+        )
+    }
+
+    private fun scramblePyraminx(): DomainScramble {
+        val puzzle = PyraminxPuzzle()
+        val wcaScramble = puzzle.generateWcaScramble(random)
+
+        return DomainScramble(
+            wcaScramble,
+            puzzle.drawScramble(wcaScramble, puzzle.defaultColorScheme),
+            PuzzleType.PYRAMINX
+        )
+    }
+
+    private fun scrambleSkweb(): DomainScramble {
```

Comment:
All of those puzzles implement Puzzle interface. Maybe remove all of those puzzle methods (scrambleTwoByTwo, scrambleThreeByThree...) and reduce them to one method that has parameter of type Puzzle. Then, in generateScrambleAndSvg we could do something like this 
``
when (puzzleType) {
    ...
    PuzzleType.TWO_BY_TWO -> scramblePuzzle(TwoByTwoCubePuzzle())
    ... 
``

Reply:
Done!